nilearn/input_data/nifti_masker.py

```diff
@@ -12,6 +12,7 @@
 from .. import _utils
 from .. import image
 from .. import masking
+from ..reporting import ReportMixin
```

Comment:
reporting imports `plotting`, which imports `matplotlib`, causing the failure when matplotlib is not installed, We will need to move `HTMLDocument` (and anything that can be useful for reporting) outside of plotting/js_plotting_utils so that other modules don't need to (indirectly) import plotting

Reply:
but actually we can probably get rid of the mixin and replace it with a `generate_report` function, which calls the `_reporting` private methods. this way any reporting-related imports and code can be contained in `_reporting` methods

Reply:
although some estimators might need to override `generate_report` altogether, which would be in favour of the `ReportMixin` solution. so let's keep everything almost the way it is, but move `HTMLDocument` and its helpers out of the `plotting` package

Reply:
Tried it in https://github.com/nilearn/nilearn/pull/2019/commits/975e626f6250bff48edd9ce1b2e2d19b63dc0545 ! Works locally, but let's see what the CI says...

Reply:
Current PEP8 Guidelines recommend absolute imports & we are moving to that for all new lines of code as well.

Reply:
OK, it's failing in the CI for several reasons, but one is that we're including `.generate_report` in the tests, which is invoking MPL through the (new) internal `._reporting` function in NiftiMasker.
It's not tied to the HTMLDocument, either -- it's just because we're calling `plotting.plot_img` to generate the actual plots that are shown in the report.  We definitely need that functionality _somewhere_. 

I think we had a discussion a few weeks (months) back, and decided to revert to `ReportMixin`. It does come with this headache if reporting is enabled, however. WDYT ? Should I make the default of reporting `False` ?

Reply:
> Current PEP8 Guidelines recommend absolute imports & we are moving to that for all new lines of code as well.

Fixed in my local copy ! I'm hesitant to keep pushing, though, until we make a decision to keep `HTMLDocument` in its new home or move it back. Mostly because I'm lazy and don't love rebasing lots of commits :)

Reply:
I don't love rebasing lots of commits because I am not a masochist. :-)

Reply:
Would skipping tests if MPL is not present work?

Reply:
I'm currently thinking just adding a `try / except ImportError` block around the actual plotting. We can return display as `None` if MPL is not available and it will generate a blank report. 

I'll move forward with integrating `HTMLDocument` and `HTMLReport`, then.

Reply:
still you would get an import error if NiftiMasker inherits from ReportMixin, if ReportMixin imports HTMLDocument from plotting, since plotting imports matplotlib in its `__init__.py` . moving `HTMLDocument` to reporting as you did seems like the least painful solution